src/common/snippets/src/lowered/linear_ir.cpp

```diff
@@ -70,11 +70,24 @@ void LinearIR::serialize(const std::string& xml, const std::string& bin) {
     auto first_node = std::make_shared<ov::op::v0::Parameter>(element::f32, Shape{});
     first_node->set_friendly_name("Start");
     first_node->get_rt_info()["execTimeMcs"] = 0;
-    std::shared_ptr<Node> body_node = first_node;
+    std::shared_ptr<Node> serialization_node = first_node;
+
+    // This map allows to get LoopBegin serialization node by original LoopBegin node
+    // It is used to draw an edge between LoopBegin and LoopEnd serialization nodes
+    std::map<snippets::op::LoopBegin*, std::shared_ptr<Node>> loops_map;
     for (const auto& expr : m_expressions) {
-        body_node = std::make_shared<op::SerializationNode>(body_node, expr);
+        const auto node = expr->get_node();
+        if (auto loop_end = ov::as_type_ptr<snippets::op::LoopEnd>(node)) {
+            auto loop_begin_serialization_node = loops_map.at(loop_end->get_loop_begin().get());
```

Comment:
Minor: `at` may throw an exception if `LoopBegin` node is not found in the map. This can't happen in a valid LIR, but it's always better to have a custom exception with an explanatory message than just a standard out_of_range. So I would add smth like
```suggestion
OPENVINO_ASSERT(loops_map.count(loop_end->get_loop_begin().get()), std::string("Can't find LoopBegin that corresponds to LoopEnd with friendly name " + loop_end->get_friendly_name()));
  auto loop_begin_serialization_node = loops_map.at(loop_end->get_loop_begin().get());
```
I don't insist though

Reply:
Good suggestion, I added the assert with the explanatory message